### PR TITLE
validations, model specs, and n+1 problems

### PR DESCRIPTION
**Pull request by algerina**
In this milestone:

- I Added the following validations:
    For the User model:
       The name must be not blank
       PostsCounter must be an integer greater than or equal to zero.
    For the Post model:
       The title must not be blank.
        The title must not exceed 250 characters.
        CommentsCounter must be an integer greater than or equal to zero.
        LikesCounter must be an integer greater than or equal to zero.

- Added unit specs for all of the models' methods and validations.

- Added flash messages in the create actions in all controllers.
ror1

ror2

ror3

ror4

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -14,4 +14,8 @@ def update_posts_counter
   def most_recent_comments
     comments.order(created_at: :desc).limit(5)
   end
+
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, presence: true, numericality: { only_integer: true }
```
Comment (Lordkaito) [inline app/models/post.rb]: - [x] Here I think you are missing the `greater_than_or_equal_to: 0` as you need to check for it as well, so please fix that issue.

**Review by Lordkaito — CHANGES_REQUESTED** [review]
Hi @algerina, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (algerina) [reply app/models/post.rb]: @lordkaito Thank you for your feedback

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @algerina

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**